### PR TITLE
UX improvements

### PR DESCRIPTION
**Pull request by smnbbrv**
This implements the feature 1 from #24

In addition to the current way of setting up the "rest actions" it is possible now to properly set up the types, furthermore it allows to avoid empty functions like `return null`.

Look how awesome it is from now on:

```ts
import { Resource, ResourceAction, ResourceParams, ResourceProcessor, TypedResourceResult } from 'ng2-resource-rest';
import { RequestMethod } from '@angular/http';

@ResourceParams({
  path: '/api/v1/preview/{id}'
})
export class PreviewResource extends Resource {

  // "new" way
  @ResourceAction({ method: RequestMethod.Get })
  public read1: ResourceProcessor<{ id: string }, { value: string }>;

  // "legacy" way
  @ResourceAction({ method: RequestMethod.Get })
  public read2(data?: { id: string; }): TypedResourceResult<{ value: string }> { return null; }

}
```

No breaking changes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,5 @@
 atlassian-ide-plugin.xml
 com_crashlytics_export_strings.xml
 crashlytics.properties
-crashlytics-build.properties
\ No newline at end of file
+crashlytics-build.properties
+typings
```
Comment (smnbbrv) [inline .gitignore]: added typings because 'node' typings were missing -> I was forced to install them locally

File: index.d.ts [hunk 2]
```diff
@@ -31,6 +31,10 @@ export interface ResourceResult {
 }
 export interface ArrayResourceResult<T> extends ResourceResult, Array<T> {
 }
+type TypedResourceResult<R extends {}> = R & {
```
Comment (smnbbrv) [inline index.d.ts]: added type as shown at #24

File: index.d.ts [hunk 3]
```diff
@@ -49,7 +53,10 @@ export declare class Resource {
     remove(data?: any, callback?: Function): ResourceResult;
     delete(data?: any, callback?: Function): ResourceResult;
 }
-export declare function ResourceAction(action?: ResourceActionBase): (target: Resource, propertyKey: string, descriptor: PropertyDescriptor) => void;
+export declare function ResourceAction(action?: ResourceActionBase): (target: Resource, propertyKey: string, descriptor?: PropertyDescriptor) => void;
```
Comment (smnbbrv) [inline index.d.ts]: adapted interface to be apply-able to the properties

File: index.d.ts [hunk 4]
```diff
@@ -49,7 +53,10 @@ export declare class Resource {
...
+export declare interface ResourceProcessor<I, O> {
```
Comment (smnbbrv) [inline index.d.ts]: added ResourceProcessor shortcut for a property-kind ResourceAction

File: index.ts [hunk 5]
```diff
@@ -418,8 +418,15 @@ export function ResourceAction(action?: ResourceActionBase) {
 
 			return ret;
 
-		}
+		};
 
+		if (descriptor) {
```
Comment (smnbbrv) [inline index.ts]: implements the logic to support both member function && member property

**Comment by troyanskiy** [comment]
Thanks a lot!

**Comment by smnbbrv** [comment]
Thank you for supporting it 👍  Would you update the current docs to reflect the changes there?

**Comment by troyanskiy** [comment]
@smnbbrv Ohh... Yes... Should.. But have no idea when... I don't really have time for that now.
I will really appreciate if you could do that if you wish and have a time for that.
Thank you very much for you help!

**Pull request merged**